Kitodo/src/main/java/org/kitodo/services/data/TemplateService.java

```diff
@@ -260,6 +276,33 @@ private void convertRelatedJSONObjects(JsonObject jsonObject, TemplateDTO templa
         return searcher.findDocuments(query.toString(), sort, offset, size);
     }
 
+    /**
+     * Get diagram image for current template.
+     *
+     * @return diagram image file
+     */
+    public InputStream getTasksDiagram(String fileName) {
+        if (Objects.nonNull(fileName) && !fileName.equals("")) {
+            File tasksDiagram = new File(ConfigCore.getKitodoDiagramDirectory(), fileName + ".svg");
+            try {
+                return new FileInputStream(tasksDiagram);
```

Comment:
Instead of returning a possible open stream maybe returning a `byte[]` value of the read file did work too? An empty `byte[]` value should returned if given file is not found.

Reply:
I would stay at the moment with this as it is.

Reply:
Ok, but this should be evaluated in future.

Reply:
That's true. In general the way of displaying this image should be checked out by frontend developers.